[CI] ProxyConnectionStrategyTests.testProxyStrategyWillOpenNewConnectionsOnDisconnect failure

**Build scan**:

https://gradle-enterprise.elastic.co/s/ajeqg2hvoymiw

**Repro line**:

./gradlew ':server:test' --tests "org.elasticsearch.transport.ProxyConnectionStrategyTests.testProxyStrategyWillOpenNewConnectionsOnDisconnect" \
  -Dtests.seed=56B58603749E5EB \
  -Dtests.security.manager=true \
  -Dtests.locale=es-BO \
  -Dtests.timezone=Asia/Tel_Aviv \
  -Druntime.java=11

**Reproduces locally?**:

No

**Applicable branches**:

master

**Failure history**:

Only once so far

**Failure excerpt**:

```
java.lang.AssertionError: expected:<6> but was:<4>
	at __randomizedtesting.SeedInfo.seed([56B58603749E5EB:E862330BC75FF383]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.elasticsearch.transport.ProxyConnectionStrategyTests.lambda$testProxyStrategyWillOpenNewConnectionsOnDisconnect$8(ProxyConnectionStrategyTests.java:156)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
```

--- a/server/src/test/java/org/elasticsearch/transport/ProxyConnectionStrategyTests.java
+++ b/server/src/test/java/org/elasticsearch/transport/ProxyConnectionStrategyTests.java
@@ -19,6 +19,8 @@ import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.transport.TransportAddress;
 import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.junit.annotations.TestIssueLogging;
+import org.elasticsearch.test.junit.annotations.TestLogging;
 import org.elasticsearch.test.transport.MockTransportService;
 import org.elasticsearch.threadpool.TestThreadPool;
 import org.elasticsearch.threadpool.ThreadPool;
@@ -98,6 +100,9 @@ public class ProxyConnectionStrategyTests extends ESTestCase {
         }
     }
 
+    @TestLogging(
+        value = "org.elasticsearch.transport.ClusterConnectionManager:TRACE,org.elasticsearch.transport.ProxyConnectionStrategy:TRACE",
+        reason = "to ensure that connections are logged")
     public void testProxyStrategyWillOpenNewConnectionsOnDisconnect() throws Exception {
         try (MockTransportService transport1 = startTransport("node1", Version.CURRENT);
              MockTransportService transport2 = startTransport("node2", Version.CURRENT)) {

--- a/server/src/test/java/org/elasticsearch/transport/ProxyConnectionStrategyTests.java
+++ b/server/src/test/java/org/elasticsearch/transport/ProxyConnectionStrategyTests.java
@@ -19,7 +19,6 @@ import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.transport.TransportAddress;
 import org.elasticsearch.test.ESTestCase;
-import org.elasticsearch.test.junit.annotations.TestIssueLogging;
 import org.elasticsearch.test.junit.annotations.TestLogging;
 import org.elasticsearch.test.transport.MockTransportService;
 import org.elasticsearch.threadpool.TestThreadPool;

--- a/server/src/test/java/org/elasticsearch/transport/ProxyConnectionStrategyTests.java
+++ b/server/src/test/java/org/elasticsearch/transport/ProxyConnectionStrategyTests.java
@@ -100,6 +100,7 @@ public class ProxyConnectionStrategyTests extends ESTestCase {
         }
     }
 
+    // This test has failed once or twice in the past. This is enabled in case it were to fail again.
     @TestLogging(
         value = "org.elasticsearch.transport.ClusterConnectionManager:TRACE,org.elasticsearch.transport.ProxyConnectionStrategy:TRACE",
         reason = "to ensure that connections are logged")

